### PR TITLE
[SW-316] Encryption for external cluster communication

### PR DESCRIPTION
**Pull request by mdymczyk**
The external cluster communication was using unencrypted sockets even if the internal security parameter was passed. This PR fixes this problem. More changes might be necessary depending on the state of the SW communication.

### CODE REVIEW

File: h2o-core/src/main/java/water/ExternalFrameHandler.java [hunk 1]
```diff
@@ -70,7 +71,7 @@
     /**
      * Method which receives the {@link SocketChannel} and {@link AutoBuffer} and dispatches the request for further processing
```
Comment (jakubhava) [inline h2o-core/src/main/java/water/ExternalFrameHandler.java]: Minor thing - the link in the comment should be changed as well to `ByteChannel`

Comment (vpatryshev) [inline h2o-core/src/main/java/water/ExternalFrameHandler.java]: No tests? Was it intentional?

Comment (jakubhava) [inline h2o-core/src/main/java/water/ExternalFrameHandler.java]: Basic tests for original functionality exist and are here https://github.com/h2oai/h2o-3/blob/master/h2o-core/src/test/java/water/ExternalFrameReaderClientTest.java and here https://github.com/h2oai/h2o-3/blob/master/h2o-core/src/test/java/water/ExternalFrameWriterClientTest.java

**Comment by mdymczyk** [comment]
@mmalohlava not yet, will test today together with https://github.com/h2oai/h2o-3/pull/727

**Comment by mdymczyk** [comment]
@mmalohlava tested with SW, ran SW in external cluster mode with encryption on and off, used wireshark to check if data can be spoofed - everything looked ok (could find the data in no-encryption mode, couldn't find it with encryption on).

**Comment by mdymczyk** [comment]
@mmalohlava tested with SW, ran SW in external cluster mode with encryption on and off, used wireshark to check if data can be spoofed - everything looked ok (could find the data in no-encryption mode, couldn't find it with encryption on).

**Comment by mmalohlava** [comment]
@h2o-ops please test

**Pull request merged**